### PR TITLE
Form query

### PR DESCRIPTION
**Pull request by taylorjohnson141**
### Code Highlights:

This PR adds the mutations to create events and eras as well as queries to get a user. Also updates the calendar and week components to correctly render the calendar.

### Where should the reviewer start?

Calender.tsx
 
### Have Tests Been Added?
- [x] No.
- [ ] Yes, but all tests are not passing.
- [ ] Yes, and all are passing.

### Any background context you want to provide?

Tests pass!

### Issues:
* Addresses #25  #36 
* Closes #

### Tracking Consistency:
- [x] added appropriate labels
- [x] My code follows the code style of this project and has removed all unnecessary annotations
- [x] I have added comments on my pull request, particularly in hard-to-understand areas
- [x] looked at PR preview to check spelling, syntax, formatting, and completion

### CODE REVIEW

File: src/Components/Week/Week.tsx [hunk 1]
```diff
@@ -4,14 +4,27 @@ import './Week.css';
 type WeekProps = {
   key: number;
   index: number;
+  color: string | null;
+  name: string | null;
 };
 
 const Week = (props: WeekProps) => {
+  let color:string | null;
```
Comment (aemcdonald) [inline src/Components/Week/Week.tsx:12]: Awesome job getting the weeks to fill!

File: src/Components/RequiredForm/RequiredForm.tsx [hunk 2]
```diff
@@ -1,57 +1,73 @@
...
+  if (data) {
+     return (
+      <FormStartPage
+      updateUserId = {props.updateUserId}
+        userId={data.createUser.id}
+        userName={data.createUser.name}
+      />
+    );
+  }
+  if (error) {
```
Comment (aemcdonald) [inline src/Components/RequiredForm/RequiredForm.tsx:57]: Nice error handling here & above with adding a loading response

File: src/Components/Question/Question.tsx [hunk 3]
```diff
@@ -1,47 +1,123 @@
...
+    if (questionType === "event") {
+      mutation = gql`
+        mutation createEvent($userId: ID!, $name: String!, $date: String!) {
+          createEvent(input: { userId: $userId, name: $name, date: $date }){
+            userId
+          }
+        }
+      `;
+    }
+    if (questionType === "era") {
```
Comment (aemcdonald) [inline src/Components/Question/Question.tsx:29]: We will need to add error handling & render a loading message to the user

**Review by krogowsk531 — APPROVED** [review]
This was super difficult but so important!  Great job getting this functionality working!

**Review by aemcdonald — APPROVED** [review]
Awesome job getting the weeks to fill for events/eras!. We will need to add some error handling to Question.tsx but otherwise everything looks great.

**Pull request merged**